cmd/vicadmin/server.go

```diff
@@ -114,39 +132,184 @@ func (s *server) listenPort() int {
 	return s.l.Addr().(*net.TCPAddr).Port
 }
 
+// Enforces authentication on route `link` and runs `handler` on successful auth
+func (s *server) AuthenticatedHandle(link string, h http.Handler) {
+	s.Authenticated(link, h.ServeHTTP)
+}
+
+func (s *server) Handle(link string, h http.Handler) {
+	s.mux.Handle(link, gorillacontext.ClearHandler(h))
+}
+
+// Enforces authentication on route `link` and runs `handler` on successful auth
+func (s *server) Authenticated(link string, handler func(http.ResponseWriter, *http.Request)) {
+	defer trace.End(trace.Begin(""))
+
+	authHandler := func(w http.ResponseWriter, r *http.Request) {
+		websession, _ := store.Get(r, sessionCookieKey) // ignore error because it is okay if it doesn't exist
```

Comment:
We do want to create a new session if it does not exist. If `Save` isn't called, it's just garbage collected. If you call `Save` it goes into `store` and will be removed by Gorilla after `MaxAge` set in `Options` when creating the store.